### PR TITLE
add support for sftp custom port

### PR DESCRIPTION
**Pull request by stephenc01**
per @Xenopathic on #12878 re submitting.

### CODE REVIEW

**Comment by PVince81** [comment]
Any update ? There's still a chance to get this into 8.0

**Comment by DeepDiver1975** [comment]
> Any update ? There's still a chance to get this into 8.0

time's up -> 8.1

File: apps/files_external/lib/sftp.php [hunk 1]
```diff
@@ -37,10 +38,21 @@ public function __construct($params) {
...
 		$this->host = $params['host'];
+		
+		//deals with sftp://server example
 		$proto = strpos($this->host, '://');
 		if ($proto != false) {
 			$this->host = substr($this->host, $proto+3);
 		}
+
+		//deals with server:port
+		$HasPort = strpos($this->host,':');
```
Comment (PVince81) [inline apps/files_external/lib/sftp.php]: This should be `$hasPort` (lower case 'h')

**Comment by LukasReschke** [comment]
I'll revive this one.

**Comment by LukasReschke** [comment]
Replaced by https://github.com/owncloud/core/pull/14460

**Pull request closed without merging**